test/org/sagebionetworks/bridge/models/ClientInfoTest.java

```diff
@@ -15,229 +10,157 @@
+        // The left side stanza:
+        assertClientInfo("appName/3 (osName/4.0.0)", "appName", 3, null, "osName", "4.0.0", null, null);
         
-        info = ClientInfo.parseUserAgentString(VALID_LONG_UA_3);
-        assertEquals("Belgium", info.getAppName());
-        assertEquals(2, info.getAppVersion().intValue());
-        assertEquals("Motorola Flip-Phone", info.getDeviceName());
-        assertEquals(ANDROID, info.getOsName());
-        assertEquals("14", info.getOsVersion());
-        assertEquals("BridgeJavaSDK", info.getSdkName());
-        assertEquals(10, info.getSdkVersion().intValue());
-    }
-    
-    @Test
-    public void deprecatedLongFormCorrectlyParsed() {
-        ClientInfo info = ClientInfo.parseUserAgentString(VALID_LONG_UA_1_DEPRECATED);
-        assertEquals("Asthma", info.getAppName());
-        assertEquals(26, info.getAppVersion().intValue());
-        assertEquals(IOS, info.getOsName());
-        assertEquals("9.1", info.getOsVersion());
-        assertEquals("BridgeSDK", info.getSdkName());
-        assertEquals(4, info.getSdkVersion().intValue());
+        // Or the right side stanza. Verify all combinations of that stanza as well.
+        assertClientInfo("(deviceName; osName/4.0.0) sdkName", null, null, "deviceName", "osName", "4.0.0", "sdkName", null);
+        assertClientInfo("(deviceName; osName/4.0.0) /5", null, null, "deviceName", "osName", "4.0.0", null, 5);
+        assertClientInfo("(deviceName; osName/4.0.0) sdkName/5", null, null, "deviceName", "osName", "4.0.0", "sdkName", 5);
+        assertClientInfo("appName/1.0 (deviceName; osName/4.0.0) sdkName/5.0.0", "appName", null, "deviceName",
+                "osName", "4.0.0", "sdkName", null);
         
-        info = ClientInfo.parseUserAgentString(VALID_LONG_UA_2_DEPRECATED);
-        assertEquals("Cardio Health", info.getAppName());
-        assertEquals(1, info.getAppVersion().intValue());
-        assertEquals(IOS, info.getOsName());
-        assertEquals("9.0.2", info.getOsVersion());
-        assertEquals("BridgeSDK", info.getSdkName());
-        assertEquals(4, info.getSdkVersion().intValue());
+        // Spacing does not break the parser
+        assertClientInfo("appName / 1 ( deviceName ; osName / 4) sdkName / 5", "appName", 1, "deviceName",
+                "osName", "4", "sdkName", 5);
         
-        info = ClientInfo.parseUserAgentString(VALID_LONG_UA_3_DEPRECATED);
-        assertEquals("Belgium", info.getAppName());
-        assertEquals(2, info.getAppVersion().intValue());
-        assertEquals(ANDROID, info.getOsName());
-        assertEquals("14", info.getOsVersion());
-        assertEquals("BridgeJavaSDK", info.getSdkName());
-        assertEquals(10, info.getSdkVersion().intValue());
-    }
-    
-    @Test
-    public void cacheWorks() {
-        ClientInfo info1 = ClientInfo.fromUserAgentCache(VALID_LONG_UA_1);
-        ClientInfo info2 = ClientInfo.fromUserAgentCache(VALID_LONG_UA_1);
```

Comment:
Do we need the fromUserAgentCache() tests?

Reply:
Yes, added them back. I deleted too many tests